nibabel/tests/test_filename_parser.py

```diff
@@ -123,3 +124,20 @@ def test_splitext_addext():
     assert res == ('..', '', '')
     res = splitext_addext('...')
     assert res == ('...', '', '')
+
+
+def test__stringify_path():
+    current_directory = pathlib.Path.cwd().as_posix()
+    res = _stringify_path('fname.ext.gz')
+    assert res == 'fname.ext.gz'
+    res = _stringify_path(pathlib.Path('fname.ext.gz'))
+    assert res == 'fname.ext.gz'
+
+    home = pathlib.Path.home().as_posix()
+    res = _stringify_path(pathlib.Path('~/fname.ext.gz'))
+    assert res == str(home) + '/fname.ext.gz'
```

Comment:
Might as well be modern about this.

```suggestion
    assert res == f'{home}/fname.ext.gz'
```